ccl/backupccl: TestBackupRestoreNegativePrimaryKey failed

ccl/backupccl.TestBackupRestoreNegativePrimaryKey [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3231439&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3231439&tab=artifacts#/) on master @ [b077a2f43a63b5180560e76c4cd64d17975134f2](https://github.com/cockroachdb/cockroach/commits/b077a2f43a63b5180560e76c4cd64d17975134f2):


```
=== RUN   TestBackupRestoreNegativePrimaryKey
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey627537406
    test_log_scope.go:74: use -show-logs to present logs inline
    backup_test.go:1109: job *jobspb.Payload_SchemaChange is not a backup: &{resume_span_list:<> desc_id:53 table_mutation_id:1 format_version:2 write_timestamp:<wall_time:1627411620321654214 > }
    backup_test.go:1222: unexpected span start at primary index
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey627537406
--- FAIL: TestBackupRestoreNegativePrimaryKey (7.41s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestBackupRestoreNegativePrimaryKey PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/bulk-io nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreNegativePrimaryKey.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-8880

ccl/backupccl.TestBackupRestoreNegativePrimaryKey [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3234788&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3234788&tab=artifacts#/) on master @ [cdace509b79339ba6c60f56e1bbe5d0b261de943](https://github.com/cockroachdb/cockroach/commits/cdace509b79339ba6c60f56e1bbe5d0b261de943):


```
=== RUN   TestBackupRestoreNegativePrimaryKey
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey074680232
    test_log_scope.go:74: use -show-logs to present logs inline
    backup_test.go:1109: job *jobspb.Payload_SchemaChange is not a backup: &{resume_span_list:<> desc_id:53 table_mutation_id:1 format_version:2 write_timestamp:<wall_time:1627458776408509790 > }
    backup_test.go:1222: unexpected span start at primary index
    panic.go:632: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey074680232
--- FAIL: TestBackupRestoreNegativePrimaryKey (37.49s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestBackupRestoreNegativePrimaryKey PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/bulk-io nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreNegativePrimaryKey.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupRestoreNegativePrimaryKey [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3235402&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3235402&tab=artifacts#/) on master @ [dbbf8fb8dad07e32445f00372bd8cda9d94a0e7e](https://github.com/cockroachdb/cockroach/commits/dbbf8fb8dad07e32445f00372bd8cda9d94a0e7e):


```
=== RUN   TestBackupRestoreNegativePrimaryKey
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey363072142
    test_log_scope.go:74: use -show-logs to present logs inline
    backup_test.go:1109: job *jobspb.Payload_SchemaChange is not a backup: &{resume_span_list:<> desc_id:53 table_mutation_id:1 format_version:2 write_timestamp:<wall_time:1627477728001631184 > }
    backup_test.go:1222: unexpected span start at primary index
    panic.go:632: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey363072142
--- FAIL: TestBackupRestoreNegativePrimaryKey (25.29s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestBackupRestoreNegativePrimaryKey PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/bulk-io nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreNegativePrimaryKey.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupRestoreNegativePrimaryKey [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3281304&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3281304&tab=artifacts#/) on master @ [847514dab6354d4cc4ccf7b2857487b32119fb37](https://github.com/cockroachdb/cockroach/commits/847514dab6354d4cc4ccf7b2857487b32119fb37):


```
=== RUN   TestBackupRestoreNegativePrimaryKey
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey523269359
    test_log_scope.go:74: use -show-logs to present logs inline
    backup_test.go:1109: job *jobspb.Payload_SchemaChange is not a backup: &{resume_span_list:<> desc_id:53 table_mutation_id:1 format_version:2 write_timestamp:<wall_time:1628572803858084053 > }
    backup_test.go:1222: unexpected span start at primary index
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey523269359
--- FAIL: TestBackupRestoreNegativePrimaryKey (2.91s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestBackupRestoreNegativePrimaryKey PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/bulk-io nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreNegativePrimaryKey.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupRestoreNegativePrimaryKey [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3286103&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3286103&tab=artifacts#/) on master @ [90809c048d05f923a67ce9b89597b2779fc73e32](https://github.com/cockroachdb/cockroach/commits/90809c048d05f923a67ce9b89597b2779fc73e32):


```
=== RUN   TestBackupRestoreNegativePrimaryKey
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey240493756
    test_log_scope.go:74: use -show-logs to present logs inline
    backup_test.go:1109: job *jobspb.Payload_SchemaChange is not a backup: &{resume_span_list:<> desc_id:53 table_mutation_id:1 format_version:2 write_timestamp:<wall_time:1628659671268126256 > }
    backup_test.go:1222: unexpected span start at primary index
    panic.go:613: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey240493756
--- FAIL: TestBackupRestoreNegativePrimaryKey (4.37s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestBackupRestoreNegativePrimaryKey PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/bulk-io nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreNegativePrimaryKey.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/backupccl.TestBackupRestoreNegativePrimaryKey [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3290817&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3290817&tab=artifacts#/) on master @ [e723a85739a2e1b514f5de3221a2177a3a37d0e6](https://github.com/cockroachdb/cockroach/commits/e723a85739a2e1b514f5de3221a2177a3a37d0e6):


```
=== RUN   TestBackupRestoreNegativePrimaryKey
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey977991030
    test_log_scope.go:74: use -show-logs to present logs inline
    backup_test.go:1111: job *jobspb.Payload_SchemaChange is not a backup: &{resume_span_list:<> desc_id:53 table_mutation_id:1 format_version:2 write_timestamp:<wall_time:1628737405101294786 > }
    backup_test.go:1224: unexpected span start at primary index
    panic.go:632: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestBackupRestoreNegativePrimaryKey977991030
--- FAIL: TestBackupRestoreNegativePrimaryKey (26.32s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestBackupRestoreNegativePrimaryKey PKG=./pkg/ccl/backupccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<details><summary>Internal log</summary>
<p>

```
benesch marked as alumn{us/a}; resolving to nvanbenschoten instead
```
</p>
</details>

/cc @cockroachdb/bulk-io nvanbenschoten
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupRestoreNegativePrimaryKey.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/backupccl/backup_test.go
+++ b/pkg/ccl/backupccl/backup_test.go
@@ -942,8 +942,8 @@ func TestBackupRestoreEmpty(t *testing.T) {
 // for tables with negative primary key data caused AdminSplit to fail.
 func TestBackupRestoreNegativePrimaryKey(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 68127, "flaky test")
 	defer log.Scope(t).Close(t)
+	skip.UnderStressRace(t, "test is too slow to run under race, presumably because of the multiple splits")
 
 	const numAccounts = 1000
 

